Orm/Xtensive.Orm.Tests/Storage/AsyncQueries/AsyncExtensionsTest.cs

```diff
@@ -0,0 +1,2315 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Xtensive.Orm.Configuration;
+using Xtensive.Orm.Tests.Storage.AsyncQueries.Model;
+
+namespace Xtensive.Orm.Tests.Storage.AsyncQueries
+{
+  public class AsyncExtensionsTest : AsyncQueriesBaseTest
+  {
+    private static Task<Session> OpenSessionAsync(Domain domain, bool isClientProfile) =>
+      domain.OpenSessionAsync(
+        new SessionConfiguration(isClientProfile ? SessionOptions.ClientProfile : SessionOptions.ServerProfile));
+
+    private static TransactionScope OpenTransactionAsync(Session session, bool isClientProfile) =>
+      isClientProfile ? null : session.OpenTransaction();
+
+    // All
+
+    [Test]
+    [TestCase(true)]
+    [TestCase(false)]
+    public async Task AllAsyncExtensionTest(bool isClientProfile)
+    {
+      await using var session = await OpenSessionAsync(Domain, isClientProfile);
+      await using (OpenTransactionAsync(session, isClientProfile)) {
```

Comment:
`ConfigureAwait` in these tests

Reply:
There is o need to do this in test code in case we don't synchronously wait for async method completion.